pexpect/pty_spawn.py

```diff
@@ -430,11 +452,7 @@ def read_nonblocking(self, size=1, timeout=-1):
         # For this case, I test isalive() before doing any reading.
         # If isalive() is false, then I pretend that this is the same as EOF.
```

Comment:
> This comment appears to be saying that on Solaris, read on a pty after EOF will block

Actually, if you read this code:
```python
       if not self.isalive():
            # timeout of 0 means "poll"
            r, w, e = self.__select([self.child_fd], [], [], 0)
            if not r:
                self.flag_eof = True
                raise EOF('End Of File (EOF). Braindead platform.')
```
you see that `select()` indicates that `child_fd` is not ready to read. So I think that the new code is safe.